### ISSUE TITLE
Ubuntu Server 24.04 ARM64 Stuck at Grub

### ISSUE THREAD

**Issue by ghostplant** [utmapp/UTM#6815]
**Describe the issue**  
I downloaded Ubuntu Server 24.04.1 ARM64 ISO from official Ubuntu website, creating a new VM in UTM, selecting Display driver to be "virtio-ramfb-gl (GPU supported)".

I can boot the ISO and see the Grub2 interface, but after go to the next step for installation, graphic stucks at this page:

<img width="797" alt="image" src="https://github.com/user-attachments/assets/c5e45d1c-ac5b-479f-94a9-616baac9ba60">

**Configuration**  
* UTM Version: 4.5.4
* macOS Version: 15.1
* Mac Chip (Intel, M1, ...):  Apple M2

**Comment by martincerven** [comment]
I think I used [noble-server-cloudimg-arm64.img](https://cloud-images.ubuntu.com/noble/current/noble-server-cloudimg-arm64.img) and it works fine. Also with some images (20.04) you need Serial (another terminal from settings) to be turned on to proceed with install.

**Comment by ghostplant** [comment]
> I think I used [noble-server-cloudimg-arm64.img](https://cloud-images.ubuntu.com/noble/current/noble-server-cloudimg-arm64.img) and it works fine. Also with some images (20.04) you need Serial (another terminal from settings) to be turned on to proceed with install.

OK, it's strange that the Ubuntu official ISO-format Server works fine for VirtualBox for Mac Silicon. Can you explain more about configuring Serial (another terminal from settings)?

**Comment by martincerven** [comment]
You just see terminal output during initiaization and boot, there are these weird services (network/cups) that it looks like nothing is happening without Serial, but you can see continuation...I recommend adding them at least during setup

**Comment by ghostplant** [comment]
@martincerven If serial is enabled, is it still compatible with graphic 2D like showing GNOME desktop?

**Comment by sav-valerio** [comment]
Yes, you can still keep Serial enabled and video output would still be displayed - you might need to wait for a while though.

In my case the culprit was `NetworkManager-wait-online.service` so after logging in I was able to run `sudo systemctl mask NetworkManager-wait-online.service` and fix that.

Once in the Serial terminal you are asked to input your credentials make sure to wait 10/20 seconds more and the greeter should then be displayed.

Additionally, make sure to choose `virtio-ramfb` as the Emulated Display Cart in the VM settings.
I also tried `virtio-ramfb-gl` which is snappier but it a lot more glitchy and unstable (maybe it's still related to https://github.com/utmapp/UTM/issues/5749?)

**Comment by martincerven** [comment]
@sav-valerio  there is nice writeup on [ utm subreddit](https://www.reddit.com/r/UTMapp/comments/1fl293h/quick_overview_of_every_qemu_display_adapter_in/) and according to them:

>  virtio-gpu-gl-pci is typically your go-to card for modern Linux distros 

Which kinda seems to make sense.

@ghostplant  yeah GUI and Serials are unrelated, but if something breaks (you uninstall your ubuntu-desktop) your only choice is to have a terminal acess.

It seems that `sudo systemctl mask NetworkManager-wait-online.service` timeout is present on all [24.04](https://www.thomas-krenn.com/en/wiki/Job_systemd-networkd-wait-online.service_start_running) and it should be okay to just turn it off.

It's especially bad if u don't have terminal output so you don't see it's hanging on a weird service, and you only see black GUI.

**Comment by ghostplant** [comment]
It works, but the systemd loading speed is too slow, much slower than VirtualBox for Mac M2
